app/workers/queue_mail_worker.rb

```diff
@@ -0,0 +1,19 @@
+class QueueMailWorker
+  include Sidekiq::Worker
+  sidekiq_options :retry => false
+
+  def perform(campaign_id)
+    campaign = Campaign.find(campaign_id)
+    if campaign.errors.present?
+      logger.info "Unable to launch campaign #{campaign_id} as errors are present."
+      return false
+    end
+    campaign.update_attributes(active: true, email_sent: true)
+    blast = campaign.blasts.create(test: false)
+    victims = Victim.where("campaign_id = ? and archive = ?", campaign_id, false)
+    logger.info "Queueing #{victims.count} emails for background delivery for campaign #{campaign_id}"
+    victims.each do |target|
+      MailWorker.perform_async(campaign.id, target.email_address, blast.id, PhishingFrenzyMailer::ACTIVE)
```

Comment:
Yeah it uses the existing mail worker?
________________________________
From: Brandon McCann<mailto:notifications@github.com>
Sent: ‎15/‎10/‎2015 17:31
To: pentestgeek/phishing-frenzy<mailto:phishing-frenzy@noreply.github.com>
Cc: Meatballs1<mailto:campbell_ab@hotmail.com>
Subject: Re: [phishing-frenzy] Background the Launch Queue (#241)

> +class QueueMailWorker
> +  include Sidekiq::Worker
> +  sidekiq_options :retry => false
> +
> +  def perform(campaign_id)
> +    campaign = Campaign.find(campaign_id)
> +    if campaign.errors.present?
> +      logger.info "Unable to launch campaign #{campaign_id} as errors are present."
> +      return false
> +    end
> +    campaign.update_attributes(active: true, email_sent: true)
> +    blast = campaign.blasts.create(test: false)
> +    victims = Victim.where("campaign_id = ? and archive = ?", campaign_id, false)
> +    logger.info "Queueing #{victims.count} emails for background delivery for campaign #{campaign_id}"
> +    victims.each do |target|
> +      MailWorker.perform_async(campaign.id, target.email_address, blast.id, PhishingFrenzyMailer::ACTIVE)

What am I missing here?  Isn't there suppose to be a mail_worker.rb file?

---
Reply to this email directly or view it on GitHub:
https://github.com/pentestgeek/phishing-frenzy/pull/241/files#r42147417